File: test/dummy/app/components/dummy/ui/header_message_component.rb
# frozen_string_literal: true

class Dummy::Ui::HeaderMessageComponent < ApplicationComponent
  def message
    @message ||= if Folio::Current.site.header_message_published?
      if controller.send(:cookies)[:hiddenHeaderMessage] != cookie
        Folio::Current.site.header_message.presence
      end
    end
  end

  def cookie
    @cookie ||= Base64.urlsafe_encode64(Folio::Current.site.header_message)
  end
end

File: test/dummy/app/components/dummy/ui/header_message_component.js
window.Folio.Stimulus.register('d-ui-header-message', class extends window.Stimulus.Controller {
  static values = {
    cookie: String,
    loaded: { type: Boolean, default: false },
  }

  connect () {
    window.Folio.RemoteScripts.run('js-cookie', () => {
      this.loadedValue = true
    }, () => {})
  }

  close (e) {
    e.preventDefault()
    this.element.parentNode.removeChild(this.element)
    window.Cookies.set('hiddenHeaderMessage', this.cookieValue)
  }
})
